src/main/java/seedu/address/ui/WidgetViewOption.java

```diff
@@ -1,10 +1,95 @@
 package seedu.address.ui;
 
+import seedu.address.model.country.Country;
+
 /**
  * Represents the display option for the Widget View.
  */
-public enum WidgetViewOption {
-    CLIENT,
-    COUNTRY_NOTES,
-    NONE
+public class WidgetViewOption {
+    private static final String CLIENT = "CLIENT";
+    private static final String COUNTRY_NOTE = "COUNTRY_NOTE";
+    private static final String NONE = "NONE";
+
+    private final boolean isClient;
+    private final boolean isCountryNote;
+    private final Country country;
+
+    /**
+     * Initializes a WidgetViewOption.
+     *
+     * @param isClient True if displaying client view.
+     * @param isCountryNote True if displaying country notes view.
+     * @param country The country of the country notes to view.
+     */
+    private WidgetViewOption(boolean isClient, boolean isCountryNote, Country country) {
+        assert !(isClient && isCountryNote); // isClient and isCountryNote cannot be both true
```

Comment:
Maybe put the comment as a message
```suggestion
        assert !(isClient && isCountryNote) : "isClient and isCountryNote cannot be both true";
```